Benchmarks/Benchmarks/Predicates/Predicates.swift

```diff
@@ -0,0 +1,234 @@
+        Benchmark("Predicate #2 - 1 KeyPath variable condition") { benchmark in
+            let monster = Monster(name: "Orc", level: 80, hp: 100, mana: 0, weapon: .sword(Sword(p1: 1, p2: 2, p3: 3, p4: 4, p5: 5)))
+            let predicate = #Predicate<Monster> { monster in
+                (monster.level == 80)
+            }
+
+            benchmark.startMeasurement()
+            var matched = 0
+            for _ in benchmark.scaledIterations {
+                if try predicate.evaluate(monster) {
+                    matched += 1
+                }
+            }
+            benchmark.stopMeasurement()
+
+            guard matched == benchmark.scaledIterations.count else {
+                fatalError("Internal error: wrong number of matched monsters")
+            }
+        }
+    }
+
+    if #available(macOS 14, *) {
+        Benchmark("Predicate #3 - 1 KeyPath computed property condition") { benchmark in
+            let monster = Monster(name: "Orc", level: 80, hp: 100, mana: 0, weapon: .sword(Sword(p1: 1, p2: 2, p3: 3, p4: 4, p5: 5)))
+            let predicate = #Predicate<Monster> { monster in
+                (monster.levelComputed == 80)
+            }
+
+            benchmark.startMeasurement()
+            var matched = 0
+            for _ in benchmark.scaledIterations {
+                if try predicate.evaluate(monster) {
+                    matched += 1
+                }
+            }
+            benchmark.stopMeasurement()
+
+            guard matched == benchmark.scaledIterations.count else {
+                fatalError("Internal error: wrong number of matched monsters")
+            }
+        }
+    }
+
+    if #available(macOS 14, *) {
+        Benchmark("Predicate #4 - 1 KeyPath nested computed property condition") { benchmark in
+            let monster = Monster(name: "Orc", level: 80, hp: 100, mana: 0, weapon: .sword(Sword(p1: 1, p2: 2, p3: 3, p4: 4, p5: 5)))
+            let predicate = #Predicate<Monster> { monster in
+                (monster.weaponP1 == 1)
+            }
+
+            benchmark.startMeasurement()
+            var matched = 0
+            for _ in benchmark.scaledIterations {
+                if try predicate.evaluate(monster) {
+                    matched += 1
+                }
+            }
+            benchmark.stopMeasurement()
+
+            guard matched == benchmark.scaledIterations.count else {
+                fatalError("Internal error: wrong number of matched monsters")
+            }
+        }
+    }
+
+    if #available(macOS 14, *) {
+        Benchmark("Predicate #5 - 3 KeyPath nested computed property conditions") { benchmark in
+            let monster = Monster(name: "Orc", level: 80, hp: 100, mana: 0, weapon: .sword(Sword(p1: 1, p2: 2, p3: 3, p4: 4, p5: 5)))
+            let predicate = #Predicate<Monster> { monster in
+                ((monster.weaponP1 == 1) &&
+                 //(monster.weaponP2 == 2) &&
+                 //(monster.weaponP3 == 3) &&
```

Comment:
If these are commented out for a particular reason, can we leave a note as to why so it's clear for maintainers in the future?

Reply:
Not sure, ping @ser-0xff

Reply:
Ok, I tested enabled it and we run into this:

```
/Users/hassila/GitHub/swift-foundation/Benchmarks/Benchmarks/Predicates/Predicates.swift:17:5: error: the compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions
let benchmarks = {
    ^~~~~~~~~~~~~~
error: fatalError
```

Reply:
Will leave it in as a separate test and comment accordingly.